### ISSUE TITLE
munin-node cannot start

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#70930]
**Describe the bug**
`services.munin-node` is broken (cannot start). It seems due to a perl change in v5.30.

Ref.
https://github.com/munin-monitoring/munin/issues/1202
https://rt.perl.org/Public/Bug/Display.html?id=134194
https://github.com/Perl/perl5/pull/23/commits/15dbe1afb2073e311b185cf095649621e6a4bfbf

**To Reproduce**
```
  services = {
    munin-node.enable = true;

    munin-cron = {
      enable = true;
      hosts = ''
        [${config.networking.hostName}]
        address localhost
      '';
    };
  };
```
After activating the configuration, `systemctl --failed` shows `munin-node.service` in failed status.

**Expected behavior**
Munin would generate some plugins in /etc/munin/plugins and `munin-node.service` should be working.

**Metadata**
NixOS 19.09 (regression from 19.03).

**Comment by tohl** [comment]
temporary workaround for nixos1909

for munin-node:
<pre>
nixpkgs.config.packageOverrides = super: { # perl 5.30 breaks plugins
  munin = super.munin.override {perlPackages = super.perl528Packages;};
};
</pre>

munin-cron needs one more workaround:
<pre>
nixpkgs.config.packageOverrides = super:
  let
    nixpkgs1903 = builtins.fetchTarball {
      url = https://nixos.org/channels/nixos-19.03/nixexprs.tar.xz;
    };
  in
    { # perl 5.30 breaks plugins
      rrdtool = (import nixpkgs1903 {}).rrdtool;
    };
</pre>

both munin-node and munin-cron should have a package option

**Comment by tohl** [comment]
hmm munin-node works with that workaround

but munin-cron does not.  i had to revert to nixos1903

**Comment by corngood** [comment]
@tohl I did this:

```
    nixpkgs.config.packageOverrides = super: {
      # https://github.com/NixOS/nixpkgs/issues/70930
      # perl 5.30 breaks plugins
      munin = super.munin.override {
        perlPackages = super.perl528Packages;
        rrdtool = super.rrdtool.override {
          perl = super.perl528Packages.perl;
        };
      };
    };
```

And both munin-node and munin-cron seem to work.

**Comment by tohl** [comment]
@corngood great, thanks!

**Comment by bjornfor** [comment]
Perl 5.30.1 is out, so if we update to that we should get munin working again.

**Comment by flokli** [comment]
Perl was bumped to 5.30.1 in #73954.
Tests work in #74935, so I assume it's fixed in master.

@volth, wdyt about backporting #73954 to 19.09 to fix munin there?

**Comment by vandenoever** [comment]
This configuration has no perl error but claims that there are no nodes:
```
        services.munin-cron = {
          enable = true;
          hosts = ''
            [${config.networking.hostName}]
            address localhost
          '';
        };
        services.munin-node = {
          enable = true;
          extraAutoPlugins = map (plugin: (/. + "${builtins.unsafeDiscardStringContext pkgs.munin.out}/lib/plugins/${plugin}")) [
            "swap" "threads" "load" "acpi" "proc" "processes" "df"
          ];
        };
```

```
munin-cron.service: Main process exited, code=exited, status=1/FAILURE
munin-cron.service: Failed with result 'exit-code'.
Failed to start batch Munin master programs.
munin-cron.service: Consumed 637ms CPU time, received 100B IP traffic, sent 140B IP traffic.
```

**Comment by sbourdeauducq** [comment]
Seems to be fixed on 20.03.
